### PR TITLE
E0d/edxapp tags

### PR DESCRIPTION
**Pull request by e0d**
Cherry picked from hack2015/docker with a merge conflict and a fixup to remove a stowaway change.

### CODE REVIEW

**Comment by feanil** [comment]
All pip, npm and bundle related calls should be `install:app-requirements` instead of `install:code`.

**Comment by e0d** [comment]
@feanil second look when you have a change.  @emrob would appreciate a quick look if you have time.

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 1]
```diff
@@ -59,11 +83,17 @@
...
+    - install
+    - install:code
 
 - name: Stat each requirements file to ensure it exists
   stat: path="{{ item }}"
   with_items: "{{ edxapp_requirements_with_github_urls }}"
   register: requirement_file_stats
+  tags:
+    - install
+    - install:code
```
Comment (feanil) [inline playbooks/roles/edxapp/tasks/deploy.yml:96]: This should also be `install:app-requirements`.  It's used in the when condition of tasks that are tagged `install:app-requirements`.

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 2]
```diff
@@ -73,6 +103,9 @@
   sudo_user: "{{ edxapp_user }}"
   when: item.stat.exists
   with_items: "{{ requirement_file_stats.results }}"
+  tags:
+    - install
+    - install:code
```
Comment (feanil) [inline playbooks/roles/edxapp/tasks/deploy.yml:109]: Same as above, this should be `install:app-requirements`

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 3]
```diff
@@ -47,12 +61,18 @@
     owner={{ supervisor_user }}
     group={{ supervisor_user }}
   sudo_user: "{{ supervisor_user }}"
+  tags:
+    - install
```
Comment (feanil) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml:65]: Should this also include `manage` as a tag?

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 4]
```diff
@@ -47,12 +61,18 @@
...
+    - install:configuration
 
 - name: "add gunicorn configuration files"
   template: >
     src={{ item }}_gunicorn.py.j2 dest={{ edxapp_app_dir }}/{{ item }}_gunicorn.py
   with_items: service_variants_enabled
   sudo_user: "{{ edxapp_user }}"
+  tags:
+    - install
```
Comment (feanil) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml:74]: Should this also include manage as a tag?

Comment (e0d) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml:74]: I don't think so, manage, in my mind, was more specific to bringing services up and down.  If you need to do both, just gang the tags.

Comment (feanil) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml:74]: ok, makes sense.

File: playbooks/roles/edxapp/tasks/deploy.yml (hunk 1 above)
Comment (e0d) [inline playbooks/roles/edxapp/tasks/deploy.yml:96]: GH being dumb look down.

File: playbooks/roles/edxapp/tasks/deploy.yml (hunk 2 above)
Comment (e0d) [inline playbooks/roles/edxapp/tasks/deploy.yml:109]: GH being dumb look down.

**Comment by feanil** [comment]
:+1:

**Pull request merged**